cpp/tests/hashing/hash_test.cpp

```diff
@@ -394,6 +400,13 @@ TEST_F(SparkMurmurHash3Test, MultiValueWithSeeds)
   fixed_width_column_wrapper<int8_t> const bytes_col({0, 100, -100, -128, 127});
   fixed_width_column_wrapper<bool> const bools_col1({0, 1, 1, 1, 0});
   fixed_width_column_wrapper<bool> const bools_col2({0, 1, 2, 255, 0});
+  fixed_point_column_wrapper<__int128_t> const decimal128_col(
+    {(__int128)0,
+     (__int128)100,
+     (__int128)-100,
+     *(__int128*)"\x01\x00\x00\x18\xC3\x7F\x2F\x60\xC3\xD1\xC4\xFC\xFF\xFF\xFF\xFF",
+     *(__int128*)"\xFF\xFF\xFF\xE7\x3C\x80\xD0\x9F\x3C\x2E\x3B\x03\x00\x00\x00\x00"},
```

Comment:
This is dangerous. The string may not have the correct alignment of int128, thus this may cause undefined behavior.

Reply:
What would you suggest as an alternative for 128 bit values?

Reply:
This way is more lengthy but I believe that it's safe:
```
union myint128_t {
    __int128 ival;
    char cvals[16];
};

myint128_t x { . cvals = "\x01....." };
myint128_t x { . cvals = "\x01....." };

decimal128_col(....., x.ival, y.ival, ....);
```

Reply:
Or maybe this?
```
union myint128_t {
    __int128 ival;
    char cvals[16];
};

decimal128_col(....., static_cast<__int128>(myint128_t{ .cvals = "\x01....." }), static_cast<__int128>(myint128_t{ .cvals = "\x01....." }), ....);
```

Reply:
Here are some nice resources on 128-bit literals that might be cleaner than dealing with alignment and such explicitly:
- Single file header that adds a suffix enabling 128 bit integer literals: https://github.com/jbapple/128-bit-literals (cc: @codereport do we have/need tooling like this?)
- Related [StackOverflow question](https://stackoverflow.com/questions/63328802/unsigned-int128-literal-gcc) with notable [comment describing a workaround](https://stackoverflow.com/questions/63328802/unsigned-int128-literal-gcc#comment111983722_63328802) that constructs a 128-bit value from two 64-bit values with a cast and a shift.

Reply:
Resolved using the cast and shift method. Overarching issue regarding 128-bit integer suffix header files retained in separate comment below.